### PR TITLE
ECS Enhancements

### PR DESCRIPTION
**Pull request by anton-yurchenko**
Hey team,

# Description
This PR introduces the following enhancements:

- Lock required dependencies
- Format the module
- Use unique resource names - this will allow to deploy the service multiple times on the same cluster (for configuration tests for example) and to maintain separate definition within the same account/region
- Support user tagging - frequently used for billing and alike
- Use single definition for multiple service deployments - allows centrally maintaining configuration for all the deployments

# How Has This Been Tested?

This PR was not tested

# Checklist:
- [ ] I have updated the relevant example in the examples directory for the module.
- [ ] I have updated the relevant test file for the module.
- [ ] This change does not affect module (e.g. it's readme file change)

### CODE REVIEW

**Review by frank-ang — CHANGES_REQUESTED** [review]
Thanks for the contribution. Suggesting minor changes.

File: modules/ecs-ec2/providers.tf
Comment (simonweil) [inline modules/ecs-ec2/providers.tf:1]: This file is usually named versions.tf

File: modules/ecs-ec2/providers.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+terraform {
+  required_version = ">= 0.13.1"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 5.24.0"
+    }
+
+    cloudflare = {
```
[the remark this answers is not in the document]
Comment (anton-yurchenko) [reply modules/ecs-ec2/providers.tf]: My bad, CTRL+C/V syndrome 😉

**Comment by anton-yurchenko** [comment]
I have updated the PR as per your commentaries
Thank you

**Comment by royfur** [comment]
@anton-yurchenko I notice that there is 1 check fail since you didn't update the changelog
once you update it we will be able to merge the changes

**Pull request merged**